src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -40,7 +40,7 @@ public static async Task<FunctionMetadataResponse> ToFunctionMetadataResponse(th
                 // Properties below this comment are not present in the kudu version.
                 IsDirect = functionMetadata.IsDirect,
                 IsDisabled = functionMetadata.IsDisabled,
-                IsProxy = functionMetadata.IsProxy,
+                IsProxy = functionMetadata is ProxyFunctionMetadata,
```

Comment:
Since we'll want IsDirect and other properties to also go away, but we still need to be able to associate our own internal info with these instances for tracking, I propose we add add a "Dictionary<string, string> Tags" member to the public interface that can be used to track these types of things, and we'd have a set of extension methods for IsDirect, IsProxy, etc. that read from this.

It's either that, or we define our own subclass that we copy provider instances into, that has the properties we need. However, that'd mean we'll be flowing this subclass type around rather than base FunctionMetadata.